lib/rspec/core/hooks.rb

```diff
@@ -455,7 +455,7 @@ def register(prepend_or_append, position, *args, &block)
         def run(position, scope, example_or_group)
           return if RSpec.configuration.dry_run?
 
-          if scope == :context
+          if scope == :context && !example_or_group.class.metadata[:skip]
```

Comment:
Well, that goes to show that our test suite has some holes in it.  Would you be up for adding a test to cover that case?